apps/vmq_server/src/vmq_queue.erl

```diff
@@ -815,9 +815,10 @@ send_notification(#session{pid=Pid} = Session) ->
     vmq_mqtt_fsm:send(Pid, {mail, self(), new_data}),
     Session#session{status=passive}.
 
-cleanup_session(SubscriberId, #session{queue=#queue{queue=Q}}) ->
+cleanup_session(SubscriberId, #session{queue=#queue{queue=Q, backup=BQ}}) ->
     _ = vmq_metrics:incr_queue_unhandled(queue:len(Q)),
```

Comment:
Should we include the size of a potentially filled backup queue in the 'unhandled' counter? At this point we actually don't know if the fsm has flushed those messages or not.